src/partial.rs

```diff
@@ -2194,6 +2253,85 @@ impl Path {
     }
 }
 
+//-------------------------------------------------------------------------------------------------
+// Extending partial paths with partial paths
+
+impl PartialPath {
+    /// Attempts to concatenate two partial paths.  If the postcondition of the “left” partial path
+    /// is not compatible with the precondition of the “right” path, we return an error describing
+    /// why.
+    ///
+    /// If the left- and right-hand partial paths have any symbol or scope stack variables in
+    /// common, then we ensure that the variables bind to the same values on both sides.  It's your
+    /// responsibility to update the two partial paths so that they have no variables in common, if
+    /// that's needed for your use case.
+    #[cfg_attr(not(feature = "copious-debugging"), allow(unused_variables))]
+    pub fn concatenate(
+        &mut self,
+        graph: &StackGraph,
+        partials: &mut PartialPaths,
+        rhs: &PartialPath,
+    ) -> Result<(), PathResolutionError> {
+        if rhs.start_node != self.end_node {
```

Comment:
In https://github.com/github/stack-graphs/pull/35 I really liked your use of `let mut lhs = self`. It's a small nit but I think it helps with the readability of the code enough to suggest it 🙇‍♂️

Reply:
Done!